src/server.py

```diff
@@ -464,37 +420,97 @@ def run_seta_details_query():
+            # better delta for them in the further.
+            branch_data = TaskRequests(str(branch), 1, str(time_of_request), 5400)
+            try:
+                session.add(branch_data)
+                session.commit()
+            except Exception:
+                session.rollback()
+
+            finally:
+                session.close()
+            counter, string_time, reset_delta = 1, time_of_request, 5400
+
+        # We should update it if that branch had already been stored.
         else:
-            priority = 0
+            counter, string_time, reset_delta = branch_info[0]
+            conn = engine.connect()
+            statement = update(TaskRequests).where(
+                TaskRequests.branch == branch).value(
+                counter == counter + 1)
+            conn.execute(statement)
+
+        last_update_time = datetime.datetime.strptime(string_time, "%Y-%m-%d %H:%M:%S")
+        delta = (time_of_now - last_update_time).total_seconds()
+
+        # we should update the time recorder if the elapse time had
+        # reach the time limit of that branch.
+        if delta >= reset_delta:
+            conn = engine.connect()
+            statement = update(TaskRequests).where(
+                TaskRequests.branch == branch).value(
+                datetime == datetime)
+            conn.execute(statement)
+
+        for d in query:
+            # we only return that job if it hasn't reach the timeout limit.
+            if delta < d[4]:
+                # Due to the priority of all high value jobs is 1, and we
+                # need to return all jobs for every 5 pushes(for now).
+                if counter + 1 % d[3] != 0:
+                    jobtype.append([d[0], d[1], d[2]])
+
+    # We don't care about the timeout variable of job if it's not a taskcluster request.
+    else:
+        query = session.query(JobPriorities.platform,
+                              JobPriorities.buildtype,
+                              JobPriorities.testtype,
+                              JobPriorities.priority,
+                              ).all()
 
-    alljobs = JOBSDATA.jobtype_query()
+    # priority = 0; run all the jobs
+    if priority != 1 and priority != 5:
+        priority = 0
 
     # Because we store high value jobs in seta table as default,
     # so we return low value jobs, means no failure related with this job as default
-
-    # priority = 0; run all the jobs
     if priority == 0:
         jobtype = alljobs
     # priority =5 run all low value jobs
-    elif priority == 5:
-        low_value_jobs = [low_value_job for low_value_job in alljobs if
-                          low_value_job not in jobtype]
-        jobtype = low_value_jobs
-    # priority =1, run all high value jobs
-    elif priority == 1:
-        pass # use jobtype as a high value query
+    else:
+        joblist = [job for job in query if job[3] == priority]
```

Comment:
my thoughts here are we could do;
for p in range(1,5):
    if p % counter == 0:
        priorities.append(p)

joblist = [job for job in query if job[3] in priorities]

^ this will allow us to match multiple priorities easily (and take care of the high/low value case)